tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -1509,7 +1492,24 @@ decompress_chunk_path_create(PlannerInfo *root, CompressionInfo *info, int paral
 	path->custom_path.path.parent = info->chunk_rel;
 	path->custom_path.path.pathtarget = info->chunk_rel->reltarget;
 
-	path->custom_path.path.param_info = compressed_path->param_info;
+	if (compressed_path->param_info != NULL)
+	{
+		/*
+		 * Note that we have to separately generate the parameterized path info
+		 * for decompressed chunk path. The compressed parameterized path only
+		 * checks the clauses on segmentby columns, not on the compressed
+		 * columns.
+		 */
+		path->custom_path.path.param_info =
+			get_baserel_parampathinfo(root,
+									  info->chunk_rel,
+									  compressed_path->param_info->ppi_req_outer);
```

Comment:
This bug was hidden because in `add_segmentby_to_equivalence_class`, besides the proper EquivalenceMembers for segmentby columns, we also generated nonsensical EMs for compressed columns. They translated to impossible join clauses and prevented the formation of parameterized paths where the decompressed and compressed parameterizations would be different.